src/server/scripts/Northrend/Gundrak/boss_moorabi.cpp

```diff
@@ -125,41 +138,49 @@ class boss_moorabi : public CreatureScript
                 }
             }
 
-            void ExecuteEvent(uint32 eventId) override
+            void UpdateAI(uint32 diff) override
             {
-                switch (eventId)
+                if (!me->IsInCombat())
+                    scheduler.Update(diff);
```

Comment:
Looks cleaner to me this way, changing it requires to skip victim checks before EventMap update.